src/NATS.Client.Hosting/NatsHostingExtensions.cs

```diff
@@ -28,6 +28,11 @@ public static IServiceCollection AddNats(this IServiceCollection services, int p
                 return new NatsConnectionPool(poolSize, options, configureConnection ?? (_ => { }));
```

Comment:
Somewhat unrelated: I was about to fix the XML docs for the method and while the comment got longer and longer, I realized that it would probably better to just have two methods.

Something like:

```diff 
-- public static IServiceCollection AddNats(this IServiceCollection services, int poolSize = 1, Func<NatsOptions, NatsOptions>? configureOptions = null, Action<NatsConnection>? configureConnection = null)
++ public static IServiceCollection AddNatsConnectionPool(this IServiceCollection services, int poolSize, Func<NatsOptions, NatsOptions>? configureOptions = null, Action<NatsConnection>? configureConnection = null)
++ public static IServiceCollection AddNatsConnection(this IServiceCollection services, Func<NatsOptions, NatsOptions>? configureOptions = null, Action<NatsConnection>? configureConnection = null)
```

Reply:
I agree, it is getting long. But I kind of like a single entry point `AddNats(opts = default)` and an options lambda of sorts. That way it creates a natural progression for the API user. One can start with just `AddNats()` without being overwhelmed by overloads, then add options as and when needed. @caleblloyd might have a much better idea about this.

Reply:
I like the idea of doing 2 extension methods - `AddNats` / `AddNatsPool` - or `AddNatsConnection` / `AddNatsConnectionPool` (the latter breaking BC but we are still in Alpha)

The `NatsPool` could have more settings than just a pool size in `NatsConnectionPoolOpts`.  It'd probably even be a good idea to default that to `Environment.ProcessorCount`

Reply:
I'll leave that for a follow up PR. :)